File: klasa 2/cpp/NWD_euklides.cpp
/*
 * NWD_euklides.cpp
 */


#include <iostream>
using namespace std;

int nwd_klasyczny(int a, int b)
/*
 * nwd_euklides.cpp
 */


#include <iostream>
using namespace std;
 
 int nwd (int a, int b) {
     int licznik = 0;
     while (a > 0){
         a = a % b;
         b = b - a;
         licznik++;
         
    };
     cout << "Powtórzeń: " << licznik << endl;  
     return b;
}
 
int nwd_klas(int a, int b) {
    int licznik = 0;
    while (a != b)
    {   
        if (a > b)
            a = a - b;
        else
            b = b - a;
        licznik++;
    }
    cout << "Powtórzeń klasyczny: " << licznik << endl;
    
    return 0;
}

int main(int argc, char **argv){
    int a, b;
        cout << "Podaj a: ";
        cin >> a;
        cout << "Podaj b: ";
        cin >> b;
        cout << nwd(a, b);
        cout << nwd_klas(a, b);
    
    return 0;
}

File: klasa 2/pajton/nwd_euklides.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  nwd_euklides.py
#  
def nwd(a, b):
    licznik = 0
    while a!= b:
        if a >b:
            a= a-b
            licznik += 1
        else:
            b=b - a
            licznik +=1
    print("Powtórzeń:",licznik)
    return a
    
    
    
def main(args):
    
    a= int(input("Podaj liczbę:"))
    b= int(input("Podaj liczbę:"))
    
    print("NWD {} i {} jest {} ". format(a, b, nwd(a, b)), end='')
    
    return 0
    
if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
